Assets/GothicVR/Scripts/UI/TurnSettingDropdownController.cs

```diff
@@ -0,0 +1,52 @@
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using UnityEngine.XR.Interaction.Toolkit;
+using GVR.Phoenix.Interface;
+public class TurnSettingDropdownController : MonoBehaviour
+{
+    public GameObject textAsset;
+    public GameObject locomotionsystem;
+    public ActionBasedSnapTurnProvider snapTurn;
+    public ActionBasedContinuousTurnProvider continuousTurn;
+    // Start is called before the first frame update
+    void Awake()
+    {
+        var dropdown = transform.GetComponent<TMP_Dropdown>();
+        dropdown.options.Clear();
+
+        List<string> items = new List<string>();
+        items.Add("Snap turn");
+        items.Add("Continuous turn");
+
+        foreach (var item in items)
+        {
+            dropdown.options.Add(new TMP_Dropdown.OptionData() { text = item });
+
+        }
+        dropdown.itemText.font = GameData.I.GothicSubtitleFont;
+
+        dropdown.onValueChanged.AddListener(delegate { DropdownItemSelected(dropdown); });
+        snapTurn = locomotionsystem.GetComponent<ActionBasedSnapTurnProvider>();
+        continuousTurn = locomotionsystem.GetComponent<ActionBasedContinuousTurnProvider>();
+    }
+
+    void DropdownItemSelected(TMP_Dropdown dropdown)
+    {
+        switch (dropdown.value)
+        {
+            case 0:
+                snapTurn.enabled = true;
+                continuousTurn.enabled = false;
+                break;
+            case 1:
+                snapTurn.enabled = false;
+                continuousTurn.enabled = true;
+                break;
+            default:
+                snapTurn.enabled = true;
+                continuousTurn.enabled = false;
+                break;
+        }
```

Comment:
Super special improvement (We're already very deep in details... ;-) ):
You can make multiple switch values lead to the same path. Here the optimization would be:

```suggestion
        switch (dropdown.value)
        {
            case 1:
                snapTurn.enabled = false;
                continuousTurn.enabled = true;
                break;
            case 0:
            default:
                snapTurn.enabled = true;
                continuousTurn.enabled = false;
                break;
        }
```

Final note, but please use the one above: You can actually remove the line _case 0:_ as it's redundant with the _default:_ label.

Reply:
I am already working on an improvement in my next PR. I will realize your suggestions there.

Reply:
Commit: e4eebf5232d8da49b27dbb956590f9267fe37697 [e4eebf5]